Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs

```diff
@@ -23,6 +26,9 @@ public sealed class BindingContract
         Duration = duration;
         ExpiringAt = expiringAt;
         BindingFrom = bindingFrom;
+
+        var @event = ContractStartedBindingEvent.Create(BindingFrom, ExpiringAt);
```

Comment:
Move this event to factory method "Start" :)

Reply:
```   
 internal static BindingContract Start(Guid id, Guid customerId, TimeSpan duration, DateTimeOffset bindingFrom, DateTimeOffset? expiringAt)
{
var bindingContract = new BindingContract(id, customerId, duration, bindingFrom, expiringAt);
var @event = ContractStartedBindingEvent.Create(bindingFrom, expiringAt);
bindingContract.RecordEvent(@event);

        return bindingContract;
    } 
```

Code will looks like that. First we create contract then event and use RecordEvent. Here we use RecordEvent outside entity. For me this doesn't look natural it in my opinion it is better to  encapsulate it inside constructor. Have you any recommendation why moving it to factory method is better? What you think?

Reply:
In general I prefer to add events based on concrete command. In this case, when doing Start command, the event is generated for start of the contract.

Ctor creates contract but on the other hand it might not be started at that point (e.g. when business logic gets more complicated). That's why I thought it would be more meaningful if it is a part of factory method that is called Start

Reply:
I see your point. I don't have strong opinion, for me both options are acceptable so. I will change it.

Reply:
Fixed